src/AbstractOperations/metric_field_reductions.jl

```diff
@@ -40,7 +40,7 @@ function Reduction(avg::Average, field::AbstractField; dims)
         # Construct summand of the Average
         L⁻¹_field_dx = field * dx / L
 
-        return Reduction(sum!, L⁻¹_field_dx; dims)
+        return Reduction(sum!, L⁻¹_field_dx; condition, mask, dims)
```

Comment:
We can't wrap the operation itself? Eg

```suggestion
        return Reduction(sum!, ConditionalOperation(L⁻¹_field_dx; condition, mask), dims)
```

Then we don't have to change `Reduction` and there's no boiler plate (this implementation ties `ConditionalOperation` and `Reduction` together intimately which could be a headache later).